src/context/AppProvider/AppContext.tsx

```diff
@@ -176,6 +194,19 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           dispatch(getValidatorData.initiate(accountId, options))
           break
         case ethChainId:
+          dispatch(
+            getOpportunityData.initiate(
```

Comment:
Same thing here, we'll want to map over a collection of opportunities known ahead of time